### ISSUE TITLE
Reduce enzyme usage in unit test by 25% - [1 of 3]

### ISSUE THREAD

**Issue by Cal-L** [MetaMask/metamask-mobile#8616]
### What is this about?

Convert the list of unit tests to use `react-testing-library` instead of `enzyme`. In total there are 53 files to convert. To make PRs easier to review and more manageable, this covers 17 test files.

### Technical Details

Convert these files

```
'app/components/Views/Root/index.test.tsx',
'app/components/Views/NavigationUnitTest/TestScreen3.test.js',
'app/components/Views/NavigationUnitTest/TestScreen2.test.js',
'app/components/Views/NavigationUnitTest/TestScreen1.test.js',
'app/components/Views/ImportPrivateKeySuccess/index.test.tsx',
'app/components/Views/AccountBackupStep1B/index.test.tsx',
'app/components/UI/WebviewProgressBar/index.test.tsx',
'app/components/UI/SeedphraseModal/index.test.tsx',
'app/components/UI/PhishingModal/index.test.tsx',
'app/components/UI/FoxScreen/index.test.tsx',
'app/components/UI/FadeView/index.test.tsx',
'app/components/UI/CustomAlert/index.test.tsx',
'app/components/UI/Confetti/index.test.tsx',
'app/components/UI/Collectibles/index.test.tsx',
'app/components/UI/Button/index.test.tsx',
'app/components/UI/AnimatedSpinner/index.test.tsx',
'app/components/UI/AddCustomToken/index.test.tsx',
```

### Acceptance Criteria

Unit tests for the updated files should all pass. Snapshots may be updated

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

**Comment by timi-codes** [comment]
Hello @Cal-L 
Regarding the acceptance criterion "Snapshots should not change": There is a discrepancy between the snapshots generated by Enzyme and RTL. 

<img width="745" alt="image" src="https://github.com/MetaMask/metamask-mobile/assets/7896429/d2ff257a-fb0a-4d2e-ba63-e75eeadf4538">

**Comment by Cal-L** [comment]
> Hello @Cal-L Regarding the acceptance criterion "Snapshots should not change": There is a discrepancy between the snapshots generated by Enzyme and RTL.
> 
> For example Enzyme uses a lowercase for components name while RTL maintains the original naming convention.What do you suggest? <img alt="image" width="745" src="https://private-user-images.githubusercontent.com/7896429/307205917-d2ff257a-fb0a-4d2e-ba63-e75eeadf4538.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkzMTc4NTMsIm5iZiI6MTcwOTMxNzU1MywicGF0aCI6Ii83ODk2NDI5LzMwNzIwNTkxNy1kMmZmMjU3YS1mYjBhLTRkMmUtYmE2My1lNzVlZWFkZjQ1MzgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDFUMTgyNTUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjJhNzQ0ZDg2NTk5MzZlMGFkMjVjZWZjYjU4MTRjNWZmNGRkNzIwNTE5MTM3MTRhYjJmYWU1NDI1MmI2ZDBjZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.wOl2gI4T5TFNu0SexDSiCVopinJZwIjlqdLkHZXpVHs">

Good catch. What you mentioned make sense. I updated the acceptance criteria to mention that snapshots may change.
